### PR TITLE
RHSTOR-6022: Disable DR for managed applications

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Issue: https://issues.redhat.com/browse/RHSTOR-6022
On top of: https://github.com/red-hat-storage/odf-console/pull/1440

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
image
image

File: packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx [hunk 1]
```diff
@@ -150,45 +149,20 @@ export const AssignPolicyView: React.FC<AssignPolicyViewProps> = ({
...
       .then(() => {
-        updateContext(
-          t('New policy assigned to application.'),
-          '',
-          AlertVariant.success,
-          ModalActionContext.ASSIGN_POLICY_SUCCEEDED
+        setModalActionContext(
+          ModalActionContext.ENABLE_DR_PROTECTION_SUCCEEDED
         );
+        // switch to list policy view
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx:159]: "list policy view" ?? we don't have list policy in the modal after refactoring...

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx [hunk 2]
```diff
@@ -273,40 +305,81 @@ export const ManagePolicyView: React.FC<ManagePolicyViewProps> = ({
...
   setModalContext,
   setModalActionContext,
 }) => {
+  const { t } = useCustomTranslation();
+  const [localModalActionContext, setLocalModalActionContext] =
+    React.useState<ModalActionContext>(modalActionContext);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  React.useState<ModalActionContext>();
+  const [inProgress, setInProgress] = React.useState(false);
+  React.useState<ModalActionContext>();
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:321]: what's the use of `React.useState<ModalActionContext>()` when u r not declaring/initialising any local variable ??

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx [hunk 3]
```diff
@@ -273,40 +305,81 @@ export const ManagePolicyView: React.FC<ManagePolicyViewProps> = ({
...
   };
 
+  const onDisable = async () => {
+    setInProgress(true);
+    await Promise.all(
+      doNotDeletePVCAnnotationPromises(dataPolicyInfo.placementControlInfo)
+    )
+      .then(() =>
+        // Delete DRPC only after the adding the PVC annotation successfully
+        Promise.all(unAssignPromises(dataPolicyInfo.placementControlInfo))
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:361]: IMO we can add annotation and delete simultaneously, no need to wait, unless there is any significant concern...

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx [hunk 4]
```diff
@@ -273,40 +305,81 @@ export const ManagePolicyView: React.FC<ManagePolicyViewProps> = ({
...
+  modalActionContext,
   loaded,
   loadError,
   dispatch,
   setModalContext,
   setModalActionContext,
 }) => {
+  const { t } = useCustomTranslation();
+  const [localModalActionContext, setLocalModalActionContext] =
+    React.useState<ModalActionContext>(modalActionContext);
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:317]: ` const [localModalActionContext, setLocalModalActionContext] =
    React.useState<ModalActionContext>();`
    
 no need to set prop as a local state...

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx [hunk 5]
```diff
@@ -315,7 +388,50 @@ export const ManagePolicyView: React.FC<ManagePolicyViewProps> = ({
...
           onEdit={onEdit}
         />
+        {!!alertProps && (
+          <Alert title={alertProps.title} variant={alertProps.variant} isInline>
+            {alertProps.description}
+          </Alert>
+        )}
       </ModalBody>
+      <ModalFooter>
+        {localModalActionContext !==
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:398]: (localModalActionContext ?? modalActionContext) !==

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx [hunk 6]
```diff
@@ -315,7 +388,50 @@ export const ManagePolicyView: React.FC<ManagePolicyViewProps> = ({
...
+            onClick={() =>
+              setLocalModalActionContext(
+                ModalActionContext.DISABLE_DR_PROTECTION
+              )
+            }
+          >
+            {t('Remove disaster recovery')}
+          </Button>
+        )}
+        {localModalActionContext ===
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:413]: (localModalActionContext ?? modalActionContext) ===

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx [hunk 7]
```diff
@@ -315,7 +388,50 @@ export const ManagePolicyView: React.FC<ManagePolicyViewProps> = ({
...
+              onClick={() => setLocalModalActionContext(null)}
+            >
+              {t('Cancel')}
+            </Button>
+            <Button
+              id="confirm-disable-dr-action"
+              variant={ButtonVariant.danger}
+              onClick={onDisable}
+              isDisabled={inProgress}
+              isLoading={inProgress}
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:428]: disable and loading both at same time ??

File: packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx (hunk 1 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/app-manage-policies/assign-policy-view.tsx:159]: Yes! my bad

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:321]: This is the worst mistake I ever made :), How come no compilation error for this

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx (hunk 3 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:361]: So the expectation is to add PVC annotation before firing deletion. If we do it parallel, we dont know which request will reach the server first. If the deletion request is reached, then disabling the user will lose the PVC. it's a huge mess.

Comment (SanjalKatiyar) [reply packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx]: ack...

File: packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx [hunk 8]
```diff
@@ -269,40 +296,80 @@ export const ManagePolicyView: React.FC<ManagePolicyViewProps> = ({
...
   loadError,
   dispatch,
   setModalContext,
   setModalActionContext,
 }) => {
+  const { t } = useCustomTranslation();
+  const [localModalActionContext, setLocalModalActionContext] =
+    React.useState<ModalActionContext>(null);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  React.useState<ModalActionContext>();
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/app-manage-policies/manage-policy-view.tsx:310]:
```suggestion
```

**Pull request merged**